apps/condo/domains/miniapp/schema/B2BAppAccessRight.test.js

```diff
@@ -186,5 +203,327 @@ describe('B2BAppAccessRight', () => {
+        expect(countByUser3).toBe(1)
+    })
+
+    test('Contact', async () => {
+        const [organization] = await registerNewOrganization(user)
+
+        const [app] = await createTestB2BApp(admin)
+        await createTestB2BAppContext(admin, app, organization, { status: 'Finished' })
+        const [right] = await createTestB2BAppAccessRight(admin, serviceUser, app)
+
+        const [property] = await createTestProperty(user, organization)
+        const [contact] = await createTestContact(user, organization, property)
+
+        // B2BApp without permissions
+        const countWithoutPermissions = await Contact.count(serviceUserClient, {})
+        expect(countWithoutPermissions).toBe(0)
+        const contactWithoutPermissions = await Contact.getOne(serviceUserClient, { id: contact.id })
+        expect(contactWithoutPermissions).toBeUndefined()
+
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await createTestContact(serviceUserClient, organization, property)
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await updateTestContact(serviceUserClient, contact.id, {})
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await Contact.softDelete(serviceUserClient, contact.id)
+        })
+
+        const countByUser = await Contact.count(user, {})
+        expect(countByUser).toBe(1)
+        const contactByUser = await Contact.getOne(user, { id: contact.id })
+        expect(contactByUser.id).toBe(contact.id)
+
+        // add permissions for B2BApp
+        await updateTestB2BAppAccessRight(admin, right.id, {
+            canReadContact: true,
+            canManageContact: true,
+            canReadOrganization: true, // required permission for manage contact
+            canReadProperty: true, // required permission for manage contact
+        })
+
+        // B2BApp with permissions
+        await createTestContact(serviceUserClient, organization, property)
+
+        const countByUser2 = await Contact.count(user, { organization: { id: organization.id } })
+        expect(countByUser2).toBe(2)
+
+        const countWithPermissions = await Contact.count(serviceUserClient, {})
+        expect(countWithPermissions).toBe(2)
+        const contactWithPermissions = await Contact.getOne(serviceUserClient, { id: contact.id })
+        expect(contactWithPermissions.id).toBe(contact.id)
+
+        await updateTestContact(serviceUserClient, contact.id, {})
+        await Contact.softDelete(serviceUserClient, contact.id)
+
+        const countByUser3 = await Contact.count(user, { organization: { id: organization.id } })
+        expect(countByUser3).toBe(1)
+    })
+
+    // NOTE: currently these schemes are not added to read/manage the service user.
+    //       It is used to test the preprocessor
+    test.skip('Ticket and TicketComment', async () => {
```

Comment:
Why add test and skip it? To run locally?

Reply:
Yes, for local testing